edi_storage/models/edi_backend.py

```diff
@@ -76,3 +82,79 @@ def _component_sort_key(self, component_class):
         return (
             1 if getattr(component_class, "_storage_backend_type", False) else 0,
         ) + res
+
+    def _storage_cron_check_pending_input(self, **kw):
+        for backend in self:
+            backend._storage_check_pending_input(**kw)
+
+    def _storage_check_pending_input(self, **kw):
+        """Create new exchange records if new files found.
+
+        Collect input exchange types and for each of them,
+        check by pattern if the a new exchange record is required.
+        """
+        self.ensure_one()
+        if not self.storage_id or not self.input_dir_pending:
+            _logger.info(
+                "%s ignored: no storage and/or input directory specified.", self.name
+            )
+            return False
+
+        exchange_types = self.env["edi.exchange.type"].search(
+            self._storage_exchange_type_pending_input_domain()
+        )
+        for exchange_type in exchange_types:
+            # NOTE: this call might keep hanging the cron
+            # if the remote storage is slow (eg: too many files)
+            # We should probably run this code in a separate job per exchange type.
+            file_names = self._storage_get_input_filenames(exchange_type)
+            _logger.info(
+                "Processing exchange type '%s': found %s files to process",
+                exchange_type.display_name,
+                len(file_names),
+            )
+            for file_name in file_names:
+                # TODO: add config for job
+                self.with_delay()._storage_create_record_if_missing(
```

Comment:
can you add a job config for this?

Reply:
What do you expect here? there is no config in all the other queue jobs on edi module.

Reply:
that's the point: I've simply drafted the delayed call but we need a queue.job.function record to use it, which should be added to `edi_storage` :wink:

Reply:
Got it. Now it is done, thanks!